lib/core/src/sdk.rs

```diff
@@ -536,51 +554,85 @@ impl LiquidSdk {
-            return;
-        };
-        let mut shutdown_receiver = cloned.shutdown_receiver.clone();
-
-        tokio::spawn(async move {
+        let sync_service = cloned.sync_service.clone()?;
+        let track_realtime_sync_events_future = async move {
             let mut sync_events_receiver = sync_service.subscribe_events();
             loop {
-                tokio::select! {
-                    event = sync_events_receiver.recv() => {
-                        if let Ok(e) = event {
-                            match e {
-                                sync::Event::SyncedCompleted{data} => {
-                                    info!(
-                                      "Received sync event: pulled {} records, pushed {} records",
-                                      data.pulled_records_count, data.pushed_records_count
-                                    );
-                                    let did_pull_new_records = data.pulled_records_count > 0;
-                                    if did_pull_new_records {
-                                        subscription_handler.track_subscriptions().await;
-                                    }
-                                    cloned.notify_event_listeners(SdkEvent::DataSynced {did_pull_new_records}).await
-                                }
+                if let Ok(event) = sync_events_receiver.recv().await {
+                    match event {
+                        sync::Event::SyncedCompleted { data } => {
+                            info!(
+                                "Received sync event: pulled {} records, pushed {} records",
+                                data.pulled_records_count, data.pushed_records_count
+                            );
+                            let did_pull_new_records = data.pulled_records_count > 0;
+                            if did_pull_new_records {
+                                subscription_handler.track_subscriptions().await;
                             }
+                            cloned
+                                .notify_event_listeners(SdkEvent::DataSynced {
+                                    did_pull_new_records,
+                                })
+                                .await
                         }
                     }
-                    _ = shutdown_receiver.changed() => {
-                        info!("Received shutdown signal, exiting real-time sync loop");
-                        return;
-                    }
                 }
             }
-        });
+        };
+
+        let shutdown_receiver = self.shutdown_receiver.clone();
+        info!("Starting track-realtime-sync-events task");
+        Some(tokio::spawn(async move {
+            run_with_shutdown(
+                shutdown_receiver,
+                "Received shutdown signal, exiting real-time sync loop",
+                track_realtime_sync_events_future,
+            )
+            .await
```

Comment:
Does this not hang in the `recv` until the task is eventually aborted after the timeout?